src/auth/service/Auth.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable, Inject } from '@nestjs/common';
+import axios, { AxiosResponse } from 'axios';
+import { Routes } from 'src/routes/Routes.enum';
+import { AuthException } from '../exceptions/auth.exception';
+
+@Injectable()
+export class AuthService {
+    constructor(
+        @Inject('BASE_URL')
+        private readonly BASE_URL: string,
+
+        @Inject('AUTH0_BASE_URL')
+        private readonly AUTH0_BASE_URL: string,
+
+        @Inject('AUTH0_CLIENT_ID')
+        private readonly AUTH0_CLIENT_ID: string,
+
+        @Inject('AUTH0_CLIENT_SECRET')
+        private readonly AUTH0_CLIENT_SECRET: string,
+
+        @Inject('AUTH0_AUTHORIZE_ROUTE')
+        private readonly AUTH0_AUTHORIZE_ROUTE: string,
+
+        @Inject('AUTH0_GET_TOKEN_ROUTE')
+        private readonly AUTH0_GET_TOKEN_ROUTE: string,
+
+        @Inject('AUTH0_AUTH_API_AUDIENCE')
+        private readonly AUTH0_AUTH_API_AUDIENCE: string,
+    ) {
+        this.validateIfExisting([
+            this.BASE_URL,
+            this.AUTH0_BASE_URL,
+            this.AUTH0_CLIENT_ID,
+            this.AUTH0_CLIENT_SECRET,
+            this.AUTH0_AUTHORIZE_ROUTE,
+            this.AUTH0_GET_TOKEN_ROUTE,
+            this.AUTH0_AUTH_API_AUDIENCE,
+        ]);
+    }
+
+    public getAuthUrl(): string {
+        const queryParams: URLSearchParams = new URLSearchParams({
+            audience: this.AUTH0_AUTH_API_AUDIENCE,
+            response_type: 'code',
+            client_id: this.AUTH0_CLIENT_ID,
+            redirect_uri: `${this.BASE_URL}${Routes.AUTH_ROUTE}${Routes.AUTH_CALLBACK_ROUTE}`,
+            scope: 'offline_access',
```

Comment:
to samo tutaj

Reply:
to w ogóle może zniknie zależnie czy będziemy obsługiwać tu refresh tokeny czy nie a tu nie jestem pewny jeszcze... oczywiście można przenieść